internal/declarative/v2/reconciler.go

```diff
@@ -286,8 +302,19 @@ func (r *Reconciler) syncResources(ctx context.Context, clnt Client, obj Object,
 	status.Synced = newSynced
 
 	if hasDiff(oldSynced, newSynced) {
-		obj.SetStatus(status.WithState(shared.StateProcessing).WithOperation(ErrWarningResourceSyncStateDiff.Error()))
+		if obj.GetDeletionTimestamp().IsZero() {
+			obj.SetStatus(status.WithState(shared.StateProcessing).WithOperation(ErrWarningResourceSyncStateDiff.Error()))
+		}
 		return ErrWarningResourceSyncStateDiff
+	} else if !obj.GetDeletionTimestamp().IsZero() {
```

Comment:
this `else if` is not the else condition for `if hasDiff`

Reply:
again, here you introduce some logic I assume make things complicated, basically the logic here should be very simple, if hasDiff, we change the state of manifest CR.

now we have two cases:
1. manifest CR not under delete, then change it to `Processing`
2. manifest CR is under delete, then change it to `Deleting`

What's the reason here to introduce ready check? here no matter if module CR exists or not, it doesn't affect the state of manifest CR.

The ready check should also just do once in each reconciliation to minimum the request to cluster.